ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileReplacer.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test;
+
+import static io.confluent.ksql.test.ExpectedTopologiesTestLoader.CONFIG_END_MARKER;
+import static io.confluent.ksql.test.ExpectedTopologiesTestLoader.SCHEMAS_END_MARKER;
+import static io.confluent.ksql.test.TopologyFileGenerator.baseConfig;
+import static io.confluent.ksql.test.TopologyFileGenerator.buildQuery;
+import static io.confluent.ksql.test.TopologyFileGenerator.findBaseDir;
+import static io.confluent.ksql.test.TopologyFileGenerator.getKsqlEngine;
+import static io.confluent.ksql.test.TopologyFileGenerator.getServiceContext;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import io.confluent.ksql.engine.KsqlEngine;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.test.tools.FakeKafkaService;
+import io.confluent.ksql.test.tools.TestCase;
+import io.confluent.ksql.test.tools.TopologyAndConfigs;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Replaces the topology part  (not config and schema headers) of the expected topologies with
+ * topologies generated by building the queries. It does not create new files, rather only updates
+ * existing files if the topologies differ from the ones created by the current code.
+ * To generate topologies of older version, the code uses the config and schema headers found in
+ * the already existing files of expected topologies.
+ *
+ * In order to use it, you need to provide the version.
+ */
+public class TopologyFileReplacer {
```

Comment:
How about I change the `TologyFileRewriter` to be abstract and then I can have the class I created extend it? This way, we have a template abstract class for how people should implement classes that modify the topologies.

Reply:
Can't you just combine the two classes? TopologyFileRewriter can be changed any way you like. It's not used by anything. It's just their as a tool to hack about with.  It obviously didn't do exactly what you needed. So is it possible to enhance it so that it has the ability to do what you needed while also maintaining the current functionality?  It's not a biggie.  If they are doing fundamentally different things then keep to classes, but link them via java docs. By that I mean in the class javadocs you might say 'this class can do x & y, if you need z then use this other class'